### PR TITLE
Extend CodeQL3000 in signalr-SignalR-official-build

### PR DESCRIPTION
**Pull request by dougbu**
Fix official builds
- exclude intermediate / unsigned files from SignCheck

Move to latest KoreBuild

nit: Ignore global.json file

Remove unnecessary auto-injected build steps

Grab binary logs in CI builds
- upload log files even in builds from forks

Extend CodeQL3000 in signalr-SignalR-official-build
- add a new biweekly schedule to keep database and tools up to database
- add CodeQL3000 variables
- add tag indicating CodeQL3000 tasks did useful work
  - ignore no-op task executions
- add a tsaoptions.json file

### CODE REVIEW

**Comment by dougbu** [comment]
[internal build #20221030-05](https://dev.azure.com/dnceng/internal/_build/results?buildId=2032766&view=results) was mostly successful. Just failed due to a flaky test.

[earlier build #20221030-03](https://dev.azure.com/dnceng/internal/_build/results?buildId=2032737&view=results) did real `CodeQL3000` work.

Both of these builds included an extra commit to add the `CodeQL3000` tasks in the YAML. This isn't necessary for builds of main and I backed that out before opening this PR. But, please let me know if there's another branch that needs to be analyzed going forward.

File: .vsts-pipelines/builds/ci-official.yml [hunk 1]
```diff
@@ -22,6 +31,15 @@ jobs:
       TeamName: AspNetCore
       OfficialBuild: true
       VisualStudioMSBuildx86Path: 'C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Bin\MSBuild.exe'
+      # CodeQL3000 tasks are auto-injected into official builds of main. Variables below are ignored otherwise.
+      # Enable CodeQL3000 auto-injected tasks. These tasks will no-op if two builds run w/in the same week.
+      Codeql.Cadence: 168
```
Comment (wtgodbe) [inline .vsts-pipelines/builds/ci-official.yml:36]: Why 168?

**Review by wtgodbe — APPROVED** [review]

File: .vsts-pipelines/builds/ci-official.yml (hunk 1 above)
Comment (dougbu) [reply .vsts-pipelines/builds/ci-official.yml:36]: One week. I mentioned that value later in the "Action Required: Enable CodeQL3000 in your pipelines ..." thread for builds that don't run frequently. Should make the database seem more "current" more of the time.

Comment (wtgodbe) [reply .vsts-pipelines/builds/ci-official.yml:36]: Could you add a comment here mentioning that it'll run weekly?

Comment (dougbu) [reply .vsts-pipelines/builds/ci-official.yml:36]: Isn't "These tasks will no-op if two builds run w/in the same week." sufficient❔

File: .config/tsaoptions.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+  "areaPath": "DevDiv\\ASP.NET Core",
```
Comment (dougbu) [inline .config/tsaoptions.json:2]: @BrennanConroy @halter73 what is the correct area for this repo❔

**Comment by dougbu** [comment]
Are problems w/ port registrations common in tests of this repo❔ I saw something similar in at least one internal test build.

``` text
System.Net.HttpListenerException : Failed to listen on prefix 'http://localhost:9392/' because it conflicts with an existing registration on the machine.
```

File: .config/tsaoptions.json (hunk 2 above)
Comment (dougbu) [reply .config/tsaoptions.json:2]: This is the main outstanding question before we merge. Would also appreciate your 👀 on this 😀

Comment (BrennanConroy) [reply .config/tsaoptions.json:2]: Ask a manager, no idea.

Comment (dougbu) [reply .config/tsaoptions.json:2]: Alright @rafikiassumani-msft or @danmoseley, what area path do you suggest for legacy SignalR❔

Comment (mkArtakMSFT) [reply .config/tsaoptions.json:2]: DevDiv\ASP.NET OOB\SignalR

Comment (dougbu) [reply .config/tsaoptions.json:2]:
```suggestion
  "areaPath": "DevDiv\\ASP.NET OOB\\SignalR",
```

**Comment by dougbu** [comment]
@BrennanConroy please merge once CI is green. I lack rights here.

**Pull request merged**